packages/@react-aria/select/src/useSelect.ts

```diff
@@ -104,18 +111,46 @@ export function useSelect<T>(props: AriaSelectOptions<T>, state: SelectState<T>,
         triggerProps['aria-label'] && !triggerProps['aria-labelledby'] ? triggerProps.id : null,
         valueId
       ].filter(Boolean).join(' '),
-      onFocus() {
+      onFocus(e: FocusEvent) {
+        if (state.isFocused) {
+          return;
+        }
+
+        if (props.onFocus) {
+          props.onFocus(e);
+        }
+
         state.setFocused(true);
       },
-      onBlur() {
+      onBlur(e: FocusEvent) {
+        if (state.isOpen) {
+          return;
+        }
+
+        if (props.onBlur) {
+          props.onBlur(e);
+        }
+
         state.setFocused(false);
-      }
+      },
+      onKeyDown: onKeyDown,
+      onKeyUp: props.onKeyUp
     }),
     valueProps: {
       id: valueId
     },
     menuProps: {
       ...menuProps,
+      onBlur: (e) => {
+        if (e.relatedTarget === ref.current || popoverRef?.current?.contains(e.relatedTarget as HTMLElement)) {
```

Comment:
Can we avoid the need for `popoverRef` altogether tight `e.currentTarget.contains(e.relatedTarget)`?